indico/modules/rb_new/client/js/components/modals/BlockingModal.jsx

```diff
@@ -44,124 +51,203 @@ function validate({period, reason, rooms}) {
+                                        endDate: moment(values[1], 'YYYY-MM-DD').format('YYYY-MM-DD')
+                                    });
+                                }
                             }}
-                            required
-                            {...props} />
+                            required={mode !== 'read'} />
         );
     };
 
     renderRoomSearchField = ({input, ...props}) => {
-        const {rooms} = this.props;
+        const {blocking: {blocked_rooms: blockedRooms}} = this.props;
+        const {mode} = this.state;
+        let label;
+
+        if (mode === 'new') {
+            label = Translate.string('Rooms to block');
+        } else {
+            label = Translate.string('Blocked rooms');
+        }
         return (
-            <ReduxFormField input={input}
+            <ReduxFormField {...props}
+                            input={input}
                             as={RoomSelector}
-                            initialValue={rooms}
-                            label={Translate.string('Rooms to block')}
+                            initialValue={blockedRooms.map((blockedRoom) => blockedRoom.room)}
+                            label={label}
                             onChange={(values) => {
                                 input.onChange(values);
                             }}
-                            required
-                            {...props} />
+                            required={mode !== 'read'} />
         );
     };
 
     renderSubmitButton = ({hasValidationErrors, pristine, submitting, submitSucceeded}) => {
+        const {mode} = this.state;
         return (
             <Button type="submit"
                     form="blocking-form"
                     disabled={hasValidationErrors || pristine || submitSucceeded}
                     loading={submitting}
                     primary>
-                <Translate>Create a blocking</Translate>
+                {mode === 'edit' ? (
+                    <Translate>Update blocking</Translate>
+                ) : (
+                    <Translate>Create a blocking</Translate>
+                )}
             </Button>
         );
     };
 
+    renderHeaderText = () => {
+        const {mode} = this.state;
+        if (mode === 'read') {
+            return <Translate>Blocking details</Translate>;
+        } else if (mode === 'edit') {
+            return <Translate>Update blocking</Translate>;
+        } else {
+            return <Translate>Create a blocking</Translate>;
+        }
+    };
+
+    hasAllowedFieldChanged = (prevValue, nextValue) => {
+        if (!prevValue || prevValue.length !== nextValue.length) {
+            return true;
+        }
+
+        prevValue = _.sortBy(prevValue, 'id');
+        nextValue = _.sortBy(nextValue, 'id');
+        return !_.every(nextValue, (val, index) => val.id === prevValue[index].id);
```

Comment:
Not exactly, it is because when viewing a specific blocking details I initialize the modal with principals serialized by marshmallow and at the same time data inside the `PrincipalSearchField` comes from a different endpoint, thus returning slightly different data. That makes comparison fail even if we compare data for the same principal.